sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/Scan.java

```diff
@@ -38,16 +36,10 @@
 @Evolving
 public interface Scan {
 
-  /**
-   * Returns the actual schema of this data source scan, which may be different from the physical
-   * schema of the underlying storage, as column pruning or other optimizations may happen.
-   */
-  StructType readSchema();
-
   /**
    * A description string of this scan, which may includes information like: what filters are
    * configured for this scan, what's the value of some important options like path, etc. The
-   * description doesn't need to include {@link #readSchema()}, as Spark already knows it.
+   * description doesn't need to include the schema, as Spark already knows it.
```

Comment:
I would rather not remove `readSchema`. The scan should be self-describing back to Spark, and the read schema is a key piece of information. In fact, I'd like to add more methods to access other things, like pushed filters and residual filters.

Reply:
> ... like pushed filters and residual filters.

hmm, are these already available from `SupportsPushDownFilters`?

Reply:
I'm not sure this is the right direction. If we add more pushdown in the future (limit, aggregate, etc.), are we going to add methods to `Scan` every time?